92-AppendixC.Rmd

```diff
@@ -0,0 +1,312 @@
+
+```{r}
+#| label: readr-glimpse
+#| message: false
+library(dplyr)
+
+anes_dta %>% 
+  select(1:6) %>% 
+  glimpse()
+```
+
+We can confirm this label status using the `haven::is.labelled()` function.
+
+```{r}
+#| label: readr-islabelled
+haven::is.labelled(anes_dta$V200002)
+```
+
+To explore the labels further, we can use the `attributes()` function. This function provides insights into both the variable labels (`$label`) and the associated value labels (`$labels`).
+
+```{r}
+#| label: readr-attributes
+attributes(anes_dta$V200002)
+```
+
+When we import a labeled dataset using {haven}, it results in a tibble containing both the data and label information. However, this is meant to be an intermediary data structure and not intended to be the final data format for analysis. Instead, we should convert it into a regular R data frame before continuing our data workflow. There are two primary methods to achieve this conversion: (1) convert to factors or (2) remove the labels.
+
+#### Option 1: Convert the vector into a factor {-}
+
+Factors are native R data types for working with categorical data. They consist of integer values that correspond to character values, known as levels. Below is a dummy example of factors. Printing `factors` shows the four different levels in the data: `strongly agree`, `agree`, `disagree`, and `strongly disagree`.
+
+```{r}
+#| label: readr-factor
+response <- 
+  c("strongly agree", "agree", "agree", "disagree")
+
+response_levels <-
+  c("strongly agree", "agree", "disagree", "strongly disagree")
+
+factors <- factor(response, levels = response_levels)
+
+factors
+```
+
+Factors are integer vectors, though they may look like character strings. We can confirm by looking at the vector's structure:
+
+```{r}
+#| label: readr-factor-view
+glimpse(factors)
+```
+
+R's factors differ from Stata, SPSS, or SAS' labeled vectors. However, we can convert labeled variables into factors using the `as_factor()` function.
+
+```{r}
+#| label: readr-factor-create
+anes_dta %>% 
+  transmute(V200002 = as_factor(V200002))
+```
+
+The `as_factor()` function can be applied to all columns in a data frame or individual ones. Below, we convert all `<dbl+lbl>` columns into factors.
+
+```{r}
+#| label: readr-factor-glimpse
+anes_dta_factor <-
+  anes_dta %>% 
+  as_factor()
+
+anes_dta_factor %>% 
+  select(1:6) %>% 
+  glimpse()
+```
+
+#### Option 2: Strip the labels{-}
+
+The second option is to remove the labels altogether, converting the labeled data into a regular R data frame. To remove, or 'zap' the labels from our tibble, we can use the {haven} package's `zap_label()` and `zap_labels()` functions. This approach removes the labels but retains the data values in their original form.
+
+Use `zap_label()` to remove the variable labels but retain the value labels:
+
+```{r}
```

Comment:
This looked weird in the code not being evaluated so I removed the `eval: false` line. However, I'm noticing now that nothing happens here.  If the anes data doesn't have variable labels should we just state that you can do this, but really highlight `zap_labels` instead?

Reply:
ANES does have variable labels. I added an example so we can see how the labels get zapped.